### DISCUSSION TITLE
Restrict includes by Pagination

### DISCUSSION THREAD

**Discussion by ablishek** [spatie/laravel-query-builder#455] [Q&A] [answered]
Hi,

Is there any way I can restrict the includes by pagination. For example if I'm querying the User with it's Posts, I want to bring only the first 9 posts, and not the 140 posts they might have written in the past.

**Comment by ablishek** [comment]
Just FYI, I am assuming I can just add a special relationship that restricts the data being brought over that relationship, but that would require someone to switch over to a specialized API if they want to load more data.

**Comment by AlexVanderbist** [comment, marked as answer]
Hi @ablishek, we don't actually handle pagination in the package and there currently isn't an easy way in Laravel's pagination that I can think of to paginate related models. If this happens to exist, I'd be happy to accept a PR or discuss a way to add it to the QueryBuilder :) 

Good luck!
